src/pg_to_evalscript/evalscript.py

```diff
@@ -103,10 +103,24 @@ def write_ndarray_definition(self):
         return pkgutil.get_data("pg_to_evalscript", f"javascript_datacube/ndarray.js").decode("utf-8")
 
     def write_datacube_creation(self):
-        return f"let {self.initial_data_name} = new DataCube(samples, '{self.bands_dimension_name}', '{self.temporal_dimension_name}', true, {json.dumps(self.bands_metadata)}, scenes)"
+        datacube_creation = ""
+        if len(self.input_bands) > 1:
+            for datasource_with_bands in self.input_bands:
+                datasource_name = datasource_with_bands["datasource"]
+                datacube_creation += f"let {datasource_name} = new DataCube(samples.{datasource_name}, '{self.bands_dimension_name}', '{self.temporal_dimension_name}', true, {json.dumps(self.bands_metadata[datasource_name] if self.bands_metadata is not None and len(self.bands_metadata) > 0 else None)}, scenes)\n\t"
+        else:
+            datasource_with_bands = self.input_bands[0]
+            datasource_name = datasource_with_bands["datasource"]
+            datacube_creation += f"let {datasource_name} = new DataCube(samples, '{self.bands_dimension_name}', '{self.temporal_dimension_name}', true, {json.dumps(self.bands_metadata[datasource_name] if self.bands_metadata is not None and len(self.bands_metadata) > 0 else None)}, scenes)\n\t"
+
+        return datacube_creation
 
     def write_runtime_global_constants(self):
-        return f"const INPUT_BANDS = {self.input_bands};"
+        all_bands = []
+        for datasource_with_bands in self.input_bands:
+            all_bands.extend(datasource_with_bands["bands"])
+
+        return f"const INPUT_BANDS = {list(set(all_bands))};"
```

Comment:
How does it work if there are 2 different collections with same band names (advanced: representing different wavelength, e.g. S2, L8)?

If I understand correctly, code generated in `makeArrayFromSamples` creates labels for `band` dimension for all datacubes with all unique band names.

Would this be a problem in some usecase?

Reply:
> How does it work if there are 2 different collections with same band names (advanced: representing different wavelength, e.g. S2, L8)?

This should not be a problem I believe, since when each DataCube is generated, each gets its own bands metadata, so even if they have the same band names, the wavelength in this case would differ for each collection/DataCube because of different bands metadata.

> If I understand correctly, code generated in makeArrayFromSamples creates labels for band dimension for datacubes for all datasources with all unique band names, even if some datasource doesn't have that band.

I am not 100% sure what would happen if some datasource doesn't have that band, I need to rethink and try this out - will get back to you with what I figure out

Reply:
> If I understand correctly, code generated in makeArrayFromSamples creates labels for band dimension for datacubes for all datasources with all unique band names, even if some datasource doesn't have that band.

I tested this with S2L1C (B10 not present in S2L2A) and S2L2A (SCL not present in S2L1C) and it worked and generated correct results, so I think this can be solved in a separate MR in the future.

Reply:
issue for later https://github.com/openEOPlatform/openeo-pg-evalscript-converter/issues/93